src/transport/SessionManager.cpp

```diff
@@ -630,6 +633,11 @@ void SessionManager::SecureGroupMessageDispatch(const PacketHeader & packetHeade
     bool decrypted = false;
     while (!decrypted && iter->Next(groupContext))
     {
+        // Optimization to reduce number of decryption attempt
+        if (groupId != groupContext.group_id)
```

Comment:
If we always want to additionally filter on this, why don't we add that matching logic to the iterator initialization:

```
groups->IterateGroupSessions(packetHeader.GetSessionId());
```

Reply:
For ease of implementation and we might not always have the groupId.